core/src/block_processor.rs

```diff
@@ -80,6 +82,11 @@ impl BlockProcessor {
             .await
             .context("failed to get block")?;
 
+        // save the block in storage
+        if let Some(block_store) = &self.block_store {
+            block_store.save(slot, block.clone(), commitment_config.commitment).await;
```

Comment:
I think the clone of the full block can be avoided because the rest of the process use block data but doesn't own them.

Reply:
Done